src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -618,6 +628,7 @@ AdaptiveExecutor(CitusScanState *scanState)
 	bool randomAccess = true;
 	bool interTransactions = false;
 	int targetPoolSize = MaxAdaptiveExecutorPoolSize;
+	static bool hasDependedJobsInAllTree = false;
```

Comment:
What does this variable do? In general, we should not keep state in static variables except when it relates to the process itself (e.g. a counter since the start of the process).

Reply:
If we have depended jobs in any part of the tree then we know that we created temporary schemas. Therefore I have used this to determine that, if so we run the cleanup logic at the top level.

Reply:
This seems like a hack. Can you come up with a different way of doing that?

Reply:
Do you see any problem in doing the cleanup at the top level ? This also speeds up the process a bit because we use only one network round per worker to do cleanup.

Reply:
I mean that we shouldn't have to use a static/global variable for this purpose.